### PR TITLE
Updating standard-numeric-format-strings to indicate the .NET Core 2.1 behavior.

### PR DESCRIPTION
**Pull request by tannergooding**
This documents a change made in .NET Core 2.1. See https://github.com/dotnet/corefx/issues/31847 for more details.

@mairaw, @eerhardt. How do we want to indicate that this impacts .NET Core 2.1 and above, but not .NET Framework (or previous versions of .NET Core)?

### CODE REVIEW

File: docs/standard/base-types/standard-numeric-format-strings.md [hunk 1]
```diff
@@ -27,7 +27,7 @@ Standard numeric format strings are used to format common numeric types. A stand
   
 -   `xx` is an optional integer called the *precision specifier*. The precision specifier ranges from 0 to 99 and affects the number of digits in the result. Note that the precision specifier controls the number of digits in the string representation of a number. It does not round the number itself. To perform a rounding operation, use the <xref:System.Math.Ceiling%2A?displayProperty=nameWithType>, <xref:System.Math.Floor%2A?displayProperty=nameWithType>, or <xref:System.Math.Round%2A?displayProperty=nameWithType> method.  
   
-     When *precision specifier* controls the number of fractional digits in the result string, the result strings reflect numbers that are rounded away from zero (that is, using <xref:System.MidpointRounding.AwayFromZero?displayProperty=nameWithType>).  
+     When *precision specifier* controls the number of fractional digits in the result string, the result string reflects a number that is rounded to a representable result nearest to the infinitely precise result, or the result with an even least significant digit if there are two equally near representable results. (that is, using <xref:System.MidpointRounding.ToEven?displayProperty=nameWithType>).
```
Comment (rpetrusha) [inline docs/standard/base-types/standard-numeric-format-strings.md]: I'd rephrase this to reflect the differences between .NET Framework, .NET Core 2.0, and .NET Core 2.1:

- When *precision specifier* controls the number of fractional digits in the result string, the result string reflects a number that is rounded to a representable result nearest to the infinitely precise result. If there are two equally near representable results:
   - \*\*On the .NET Framework and .NET Core up to .NET Core 2.0\*\*, the result with the greater least significant digit (that is, using \<xref:System.MidpointRounding.ToEven?displayProperty=nameWithType>).
   - \*\*On .NET Core 2.1 and later\*\*, the result with an even least significant digit (that is, using \<xref:System.MidpointRounding.ToEven?displayProperty=nameWithType>).

**Review by rpetrusha — CHANGES_REQUESTED** [review]
Thanks for opening this PR, @tannergooding. I've indicated in my comment how we might document all versions of .NET Core and .NET Framework.

File: docs/standard/base-types/standard-numeric-format-strings.md [hunk 2]
```diff
@@ -27,8 +27,10 @@ Standard numeric format strings are used to format common numeric types. A stand
   
 -   `xx` is an optional integer called the *precision specifier*. The precision specifier ranges from 0 to 99 and affects the number of digits in the result. Note that the precision specifier controls the number of digits in the string representation of a number. It does not round the number itself. To perform a rounding operation, use the <xref:System.Math.Ceiling%2A?displayProperty=nameWithType>, <xref:System.Math.Floor%2A?displayProperty=nameWithType>, or <xref:System.Math.Round%2A?displayProperty=nameWithType> method.  
   
-     When *precision specifier* controls the number of fractional digits in the result string, the result strings reflect numbers that are rounded away from zero (that is, using <xref:System.MidpointRounding.AwayFromZero?displayProperty=nameWithType>).  
-  
+    When *precision specifier* controls the number of fractional digits in the result string, the result string reflects a number that is rounded to a representable result nearest to the infinitely precise result. If there are two equally near representable results:
```
Comment (tannergooding) [inline docs/standard/base-types/standard-numeric-format-strings.md]: Is there a good way to see how this will actually render? The `markdown` view looks "off" with this particular bullet-point nesting.

File: docs/standard/base-types/standard-numeric-format-strings.md (hunk 1 above)
Comment (tannergooding) [reply docs/standard/base-types/standard-numeric-format-strings.md]: Updated

File: docs/standard/base-types/standard-numeric-format-strings.md [hunk 3]
```diff
@@ -27,8 +27,10 @@ Standard numeric format strings are used to format common numeric types. A stand
...
+    - \*\*On the .NET Framework and .NET Core up to .NET Core 2.0\*\*, the result with the greater least significant digit (that is, using \<xref:System.MidpointRounding.ToEven?displayProperty=nameWithType>).
+    - \*\*On .NET Core 2.1 and later\*\*, the result with an even least significant digit (that is, using \<xref:System.MidpointRounding.ToEven?displayProperty=nameWithType>).
```
Comment (pkulikov) [inline docs/standard/base-types/standard-numeric-format-strings.md]: @rpetrusha did you mean to use **bold** with `**` syntax? Then all escape backslashes should be removed. 
@tannergooding there is also no need in the backslash in front of `<xref`

Comment (tannergooding) [reply docs/standard/base-types/standard-numeric-format-strings.md]: Fixed.

File: docs/standard/base-types/standard-numeric-format-strings.md (hunk 2 above)
Comment (rpetrusha) [reply docs/standard/base-types/standard-numeric-format-strings.md]: Yes, there is, @tannergooding. It builds on our [internal review site](https://review.docs.microsoft.com/en-us/dotnet/standard/base-types/standard-numeric-format-strings?branch=pr-en-us-7202).

**Comment by tannergooding** [comment]
image

**Review by rpetrusha — APPROVED** [review]
I've made a small change, @tannergooding, by adding a subject ("the runtime selects") to each bulleted list item. I think that this is ready to merge; let me know whether you agree.

**Comment by tannergooding** [comment]
@rpetrusha, LGTM.

**Review by eerhardt — APPROVED** [review]
Thanks @tannergooding and @rpetrusha.
